compiler/src/dotty/tools/repl/Rendering.scala

```diff
@@ -64,11 +64,24 @@ private[repl] class Rendering(parentClassLoader: Option[ClassLoader] = None) {
       myClassLoader
     }
 
+  /** Used to elide long output in replStringOf.
+   *
+   * TODO: Perhaps implement setting scala.repl.maxprintstring as in Scala 2, but
+   * then this bug will surface, so perhaps better not?
+   * https://github.com/scala/bug/issues/12337
+   */
+  private[repl] def truncate(str: String): String = {
+    val ncp = str.codePointCount(0, str.length) // to not cut inside code point
+    if ncp > MaxStringElements && ncp > 3 then
```

Comment:
Not sure if `ncp > 3` is necessary. I assume that we always have `MaxStringElements >= 3`, otherwise we could not even print the `...`.

Reply:
Otherwise looks good

Reply:
Thanks. Well in Scala 2 there is some special case where the max setting set to zero is special if I remember right. I think, if we don't implement the setting then the simplification you suggest makes sense. I'll implement that in my n ext commit.